### ISSUE TITLE
After some time, an Azure function gets stuck running to 100% CPU

### ISSUE THREAD

**Issue by PureKrome** [Azure/azure-functions-host#4599]
Hello Team,

recently (last fortnight or so) our team has noticed that our app service _plan_ was running at 100%. The plan has 2x .NET core "web api" websites and 1x Azure Function. Stopping the function caused the cpu to drop to 0%. 

The function contains a few timer based triggers and queue triggers. Currently it's _very_ low usage as there's not really any queue items getting triggered, while the time based kicks off every 5 or so minutes.

There seems to be no real obvious correlation between when the CPU spikes.

I tried to get some profiling information via KUDU using `Process -> Start Profiling -> Stop Profiling` and recevied a `.diagsession` file (which I can make available, if required). It's also really hard getting any profiling via KUDU when the App Service Plan is running at 100% 😢 

Exploring the `.diagsession` seems to suggest that the issue lies with some low level network HTTP call? and the call is to Application Insights?

I thought it might be quicker to get some help via Twitter so I [initially asked there](https://twitter.com/purekrome/status/1141117971357556736) and got some suggestions ([here](https://twitter.com/cgillum/status/1141347291090915329) and [here](https://twitter.com/brett_sam/status/1141385622885548033)) about removing some A.I. setting (ref: #4551) which I tried but [didn't seem to help](https://twitter.com/purekrome/status/1141689329237938177).

So I'm not too sure what to do from here to further debug this issue. I'm not even 100% sure if the analysis I did is accurate or not?

More info: 
- App Service Plan: B2: 1
- NuGet package: `Microsoft.NET.Sdk.Functions 1.0.28`
- Code is running in 'Release' mode (with respect to how it was built)

Here are some pics that help explain my analysis:

![image](https://user-images.githubusercontent.com/899878/59888827-dc6a8e80-940c-11e9-9683-068d95931129.png)

![image](https://user-images.githubusercontent.com/899878/59888840-e5f3f680-940c-11e9-8edc-33865bb71b36.png)

and later on .. another pic about CPU usage again..

![image](https://user-images.githubusercontent.com/899878/59888866-fd32e400-940c-11e9-9322-54eff7f15a13.png)

So now i'm going to inspect the contents of the `.diagsession` file....

![image](https://user-images.githubusercontent.com/899878/59888887-1471d180-940d-11e9-9952-f4b217a12b4a.png)

![image](https://user-images.githubusercontent.com/899878/59888894-1a67b280-940d-11e9-83d3-b834b9fd6d7f.png)

⚠️  Notice how it's stuck 3x here? like .. 3 cores or whatever are stuck at 100% each which translates to about 33% for each. I think i opened each of those up and they looked to be the same problem.

Here's one of them ....

![image](https://user-images.githubusercontent.com/899878/59888898-1fc4fd00-940d-11e9-853d-404c46760687.png)

![image](https://user-images.githubusercontent.com/899878/59888907-2784a180-940d-11e9-8601-f302f78cf7d8.png)

I've also added the following to my `hosts.json` file ....

```
{
  "version": "2.0",
  "logging": {
    "applicationInsights": {
      "httpAutoCollectionOptions": {
        "enableHttpTriggerExtendedInfoCollection" : false
      }
    }
  }
}
```

and this was last night's cpu ... 

![image](https://user-images.githubusercontent.com/899878/59888999-a083f900-940d-11e9-944d-ce253dafe070.png)

and with some information added to it ...

![image](https://user-images.githubusercontent.com/899878/59889040-c9a48980-940d-11e9-942d-bd08df6c64e7.png)

So - would love some thoughts about how to progress with this. Again, I can make the `.diagsession` available if this helps OR tell me how I can deep dive into the methods to see the code/call stack (if possible?)

Finally - I'm not confident that my quick analysis is accurate and that the issue is with A.I. but it could be my own code which I've just not read correct via the Diagnostic Session in VS.

Thanks kindly for reading this and any potential help.

**Comment by safihamid** [comment]
@PureKrome can you please share your profiles as well as your function code? 
Also is this easy to repro if we deploy your function code?
feel free to send them to hamids at microsoft (dot) com

**Comment by brettsam** [comment]
Also pinging @lmolkova. Does anything jump out at you here?

**Comment by lmolkova** [comment]
Please also add  `"enableW3CDistributedTracing" : false` under `httpAutoCollectionOptions`

**Comment by PureKrome** [comment]
Apologies for the late reply -> I've been out of the office since my previous post. I've not looked at any stats or sent data so I'll action all of this later on today or tomorrow after my important tasks have been triaged, etc. Again, sorry for the late replies.

**Comment by PureKrome** [comment]
Hmm... Ok, so here's the latest 7 days.

![image](https://user-images.githubusercontent.com/899878/60145776-774ada80-980a-11e9-8f5f-97d247a8d67a.png)

So just after I waited for ages, trying things .. and then posted this issue, it now seems to be working? Just my luck 😟 

Maybe the `enableHttpTriggerExtendedInfoCollection : false` worked? I'm going to try and _undo_ that and see what happens over the next day or so. 

I've also emailed the `diagsession` to @safihamid in case there was something important in there.

**Comment by PureKrome** [comment]
@safihamid Not sure if you had any chance to review the `.diagsession` I emailed you, but right now the CPU seems to be OK, even with that setting removed.

![image](https://user-images.githubusercontent.com/899878/60307073-5dce9d80-9986-11e9-9c91-6b94dea06468.png)

So I'll close this issue right now because the issue seems to have gone - but I'm not confident it has been fixed. If (or when!) this starts again, I'll re-open this issue and ping you lovely folks again.
